src/Generic/Data/Internal/Enum.hs

```diff
@@ -187,6 +188,53 @@ gminBound = to gMinBound
 gmaxBound :: (Generic a, GBounded (Rep a)) => a
 gmaxBound = to gMaxBound
 
+-- | Generic 'range'.
+--
+-- @
+-- import 'Data.Ix'
+-- instance 'Ix' MyType where
+--   'range' = 'grange'
+--   'index' = 'gindex'
+--   'inRange' = 'ginRange'
+-- @
+grange :: (Generic a, GIx (Rep a)) => (a, a) -> [a]
+grange (m, n) = map to $ gRange (from m, from n)
+
+-- | Generic 'index'.
+--
+-- See also 'grange'.
+gindex :: (Generic a, GIx (Rep a)) => (a, a) -> a -> Int
+gindex b i
+  | ginRange b i = gunsafeIndex b i
+  | otherwise = errorWithoutStackTrace "Error in array index"
+
+-- | Generic 'unsafeIndex'.
+--
+-- The functions unsafeIndex and unsafeRangeSize belong to 'Ix' but are
+-- internal to GHC and hence not exported from the module Data.Ix. However they
+-- are exported from the module GHC.Arr.
+-- See 'grange' for how to define an instance of 'Ix' such that it does not
+-- depend on the stability of GHCs internal API. Unfortunately this results in
+-- additional (unnecessary) bound checks.
+-- With the danger of having no stability guarantees for GHC's internal API one
+-- can alternatively define an instance of 'Ix' as
+--
+-- @
+-- import GHC.Arr
```

Comment:
```suggestion
-- import "Data.Ix"
```

Double-quotes are Haddock formatting for links to modules.

Reply:
The [doc](https://downloads.haskell.org/ghc/8.8.2/docs/html/libraries/base-4.13.0.0/GHC-Arr.html) says: "unsafeIndex is like index, but without checking that the value is in range."

Linking to GHC.Arr or unsafeIndex generates a dead link for me with GHC 8.6.5. I assume that is because GHC 8.8 is the first version that generates documentation for the `GHC.Arr` module?

Additionally `stack haddock` gives me warnings:
```
Warning: Generic.Data.Internal.Generically: could not find link destinations for:
    unsafeIndex unsafeRangeSize
Warning: Generic.Data: could not find link destinations for:
    unsafeIndex unsafeRangeSize
Warning: Generic.Data.Microsurgery: could not find link destinations for:
    unsafeIndex unsafeRangeSize
```
Do you have the same problem? As far as I can see this is the result of importing `Data.Ix`, which imports `GHC.Arr` but does not export the unsafe methods.

Reply:
I see. Let's not link to those identifiers, because they're not exported from `Data.Ix` (`GHC.*` modules are mostly meant to be internal, although `GHC.Generics` and `GHC.TypeLits` break that rule). We can write `@unsafeIndex@` instead of `'unsafeIndex'`.